src/inference/src/ie_core.cpp

```diff
@@ -2002,6 +2002,20 @@ CompiledModel Core::compile_model(const std::string& modelPath, const std::strin
     });
 }
 
+CompiledModel Core::compile_model(const std::string& model,
+                                  const ov::Tensor& weights,
+                                  const std::string& deviceName,
+                                  const AnyMap& config) {
+    InferenceEngine::Blob::Ptr blob;
+    if (weights) {
+        blob = weights._impl;
+    }
+    OV_CORE_CALL_STATEMENT(
+        auto _model = _impl->ReadNetwork(model, blob).getFunction();
```

Comment:
we need to use caching here.

Reply:
Could you give some details on cache what?

Reply:
model cache feature https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_caching_overview.html

Reply:
model caching is done in compile_model() calling, here we should set cache dir to enable it?
Do we always cache model in this interface even caller didn't explicitly set cache dir ?

Reply:
When user calls this function, he except that in case of model caching, we don't parse the model again and again on the second, third and other runs. 

Please, have a look at this implementation with filename https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/ie_core.cpp#L849-L884, you can see that in case of cache, we try to load from cache first (where compute hash based on filename), in case of we have not found anything in cache, we do perform honest read_model + compile_model.

The same should happen in your implementation:
- you compute hash based on std::string and ov::tensor, since they represent a model. It's expected to be must faster than compute hash based on ov::Model (and you have to parse ov::Model from these std::string and ov::Tensor first and it also takes time). 
- on the first run where we have cache miss, you read_model and compile_model
- on other iterations, we have cache hits

Reply:
could you please also provide a link to potential place in ONNX OV EP where this function will be used?

Reply:
This implement is to support cache model with the hash key of (std::string model and ov::tensor), is my understanding correct?

There are 2 potential places in ONNX OV EP will call read cache model:
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/openvino/backend_utils.cc#L158-L212

or

https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/openvino/backends/basic_backend.cc#L57-L76

And it need do some modification if we provide new API.

Reply:
> This implement is to support cache model with the hash key of (std::string model and ov::tensor), is my understanding correct?

Yes

> https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/openvino/backends/basic_backend.cc#L57-L76

This block of code should after be replaced with our new function:
```c++
cm = core.compile_model(str, ov::cache_dir("./dir"));
```
where:
1. first iteration compute hash, doesn't see compiled_blob by hash in cache, does honest read_model + compile internally
2. second, compute hash based on string (much faster than read_model + hash on ov::Model), sees cache entry from iteration 1, imports from cache

Intermediate steps like Constant Folding, changing precisions will also be removed since they are not needed now.

Reply:
I have got some test data(mobilenet-v3, xml_size=140KB, bin_size=13.5MB) on CPU plugin:

Original (read_model + compile_model) + enable cache:
           first call: 510ms
           second call: 326 ms

Original (read_model + compile_model) + disable cache:
           first call: 478 ms
           second call: 478 ms

Current implement(compile_model):
           first call: 554 ms
           second call: 344 ms

Reply:
![image](https://user-images.githubusercontent.com/31196718/207541125-7e9878a1-827a-468c-b6f0-f0bd652347ec.png)

Per above test result, hash key with (string mode + weights) does run faster than hash key with ov::Model, and the latency of fully load model is less if cache is hit, but the first iteration latency become a little worse.

Reply:
![image](https://user-images.githubusercontent.com/31196718/207777174-63119585-ac96-49db-9d9d-21520eab91fb.png)